starter_code/src/movies.js

```diff
@@ -1,20 +1,186 @@
+        return accumulator + currentValue;
+    });
+
+    return parseFloat(sum / rates.length);
+
+}
+
+console.log(ratesAverage(movies));
 
 // Get the average of Drama Movies
 
+function dramaMoviesRate (object) {
+    var dramaMovies = object.filter(function(item) {
+        if (item.genre.indexOf('Drama') >= 0) {
+            return item;
+        }
+    });
+
+    var rates = dramaMovies.map(function(item){
+        var rate = parseFloat(item.rate);
+        return rate;
+    });
+
+    var sum = rates.reduce(function(accumulator, currentValue) {
+        return accumulator + currentValue;
+    });
+
+    return parseFloat(sum / rates.length);
+}
+
+console.log(dramaMoviesRate(movies));
 
 // Order by time duration, in growing order
 
+function orderByDuration (object) {
+    var sortTime = object.sort(function(previous, current) {
+        var previousMovieDuration = previous.duration;
+        var currentMovieDuration = current.duration;
+        var previousMovieName = previous.title;
+        var currentMovieName = current.title;
+
+        if (previousMovieDuration == currentMovieDuration) {
+            if (previousMovieName < currentMovieName) {
+                return -1;
+            } else if (previousMovieName > currentMovieName) {
+                return 1;
+            }
+        }
+        return previousMovieDuration - currentMovieDuration;
+    });
+    
+    return sortTime;
+    }
+    
+
+console.log(orderByDuration(movies));
 
 // How many movies did STEVEN SPIELBERG
 
+function howManyMovies (object) {
+    var dramaMovies = object.filter(function(item) {
+        if (item.genre.indexOf('Drama') >= 0 && item.director == 'Steven Spielberg') {
+            return item;
+        }
+    });
+
+    var result = '';
 
+    if (dramaMovies.length == 0) {
+        result = undefined;
+    } else {
+        result = "Steven Spielberg directed " + dramaMovies.length + " drama movies!";
+    }
+    return result;
+}
+
+console.log(howManyMovies(movies));
 // Order by title and print the first 20 titles
 
+function orderAlphabetically (object) {
+    var sortAlphabetically = object.sort(function(previous, current) {
+        var previousMovieName = previous.title;
+        var currentMovieName = current.title;
+
+        if (previousMovieName < currentMovieName) {
+            return -1;
+        } else if (previousMovieName > currentMovieName) {
+            return 1;
+        }
+        return 0;
+    });
+
+    var resultArray = [];
+
+    if (sortAlphabetically.length > 20) {
+        for (var i = 0; i < 20; i++) {
+            resultArray.push(sortAlphabetically[i].title);
+        }
+    } else {
+        for (var i = 0; i < sortAlphabetically.length; i++) {
+            resultArray.push(sortAlphabetically[i].title);
+        }
+    }
 
+    
+    return resultArray;
+}
+
+console.log(orderAlphabetically(movies));
 // Best yearly rate average
+
+function bestYearAvg (object) {
```

Comment:
Para resolver os testes do jasmine:
- Quando for passado um array vazio para a função, ele deve retornar 'undefined'. 
- Quando for passado um array com apenas um elemento, ele deverá tratar essa condição.
- A função não está funcionando corretamente para um array com vários filmes.